src/Microsoft.ApiDesignGuidelines.Analyzers/Core/EquatableAnalyzer.Fixer.cs

```diff
@@ -0,0 +1,247 @@
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            SyntaxGenerator generator = SyntaxGenerator.GetGenerator(context.Document);
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            SyntaxNode declaration = root.FindNode(context.Span);
+            declaration = generator.GetDeclaration(declaration);
+            if (declaration == null)
+            {
+                return;
+            }
+
+            SemanticModel model =
+                await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            INamedTypeSymbol type = model.GetDeclaredSymbol(declaration) as INamedTypeSymbol;
+            if (type == null || type.TypeKind != TypeKind.Class && type.TypeKind != TypeKind.Struct)
+            {
+                return;
+            }
+
+            INamedTypeSymbol equatableType = WellKnownTypes.GenericIEquatable(model.Compilation);
+            if (equatableType == null)
+            {
+                return;
+            }
+
+            // We cannot have multiple overlapping diagnostics of this id.
+            Diagnostic diagnostic = context.Diagnostics.Single();
+
+            if (type.TypeKind == TypeKind.Struct && !TypeImplementsEquatable(type, equatableType))
+            {
+                context.RegisterCodeFix(new MyCodeAction(
+                    MicrosoftApiDesignGuidelinesAnalyzersResources.ImplementEquatable,
+                    async ct =>
+                        await ImplementEquatableInStructAsync(context.Document, declaration, type, model.Compilation,
+                            equatableType, ct).ConfigureAwait(false),
+                    equivalenceKey: MicrosoftApiDesignGuidelinesAnalyzersResources.ImplementEquatable), diagnostic);
+            }
+
+            if (!type.OverridesEquals())
+            {
+                context.RegisterCodeFix(new MyCodeAction(
+                    MicrosoftApiDesignGuidelinesAnalyzersResources.OverrideEqualsOnImplementingIEquatableCodeActionTitle,
+                    async ct =>
+                        await OverrideObjectEqualsAsync(context.Document, declaration, type, equatableType,
+                            ct).ConfigureAwait(false),
+                    equivalenceKey: MicrosoftApiDesignGuidelinesAnalyzersResources.OverrideEqualsOnImplementingIEquatableCodeActionTitle), diagnostic);
+            }
+        }
+
+        private bool TypeImplementsEquatable(INamedTypeSymbol type, INamedTypeSymbol equatableType)
+        {
+            INamedTypeSymbol constructedEquatable = equatableType.Construct(type);
+            INamedTypeSymbol implementation = type
+                .Interfaces
```

Comment:
Are you sure? I'm asking because the associated analyzer contains the same test. Should both be updated to use `AllInterfaces` instead?